pywhy_stats/kci.py

```diff
@@ -0,0 +1,371 @@
+        kernel_X = func:`sklearn.metrics.pairwise.pairwise_kernels.polynomial`
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    test_statistic, pvalue = _kernel_test(X, Y, None, kernel_X, kernel_Y, None, approx, null_sample_size, threshold,
+                                          normalize_data)
+    return PValueResult(pvalue=pvalue, statistic=test_statistic)
+
+
+def condind(X: np.ndarray,
+            Y: np.ndarray,
+            Z: np.ndarray,
+            kernel_X: Optional[Callable[[np.ndarray], np.ndarray]] = None,
+            kernel_Y: Optional[Callable[[np.ndarray], np.ndarray]] = None,
+            kernel_Z: Optional[Callable[[np.ndarray], np.ndarray]] = None,
+            approx: bool = True,
+            null_sample_size: int = 1000,
+            threshold: float = 1e-5,
+            normalize_data: bool = True) -> PValueResult:
+    """
+    Tests whether X and Y given Z are conditionally independent using the Kernel Independence Tests.
+
+    For testing conditional independence on continuous data, we leverage kernels :footcite:`Zhang2011` that are
+    computationally efficient.
+
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples, n_dimensions1)
+        Data for variable X, which can be multidimensional.
+    Y : ArrayLike of shape (n_samples, n_dimensions2)
+        Data for variable Y, which can be multidimensional.
+    Z : ArrayLike of shape (n_samples, n_dimensions3)
+        Data for variable Z, which can be multidimensional.
+    kernel_X : Callable expecting a numpy array as input and returns another numpy array
+        The kernel function for X. By default, the RBF kernel is used for continuous and the delta kernel for
+        categorical data.
+    kernel_Y : Callable expecting a numpy array as input and returns another numpy array
+        The kernel function for Y. By default, the RBF kernel is used for continuous and the delta kernel for
+        categorical data.
+    kernel_Z : Callable expecting a numpy array as input and returns another numpy array
+        The kernel function for Z. By default, the RBF kernel is used for continuous and the delta kernel for
+        categorical data.
+    approx : bool
+        Whether to use the Gamma distribution approximation for the pvalue, by default True.
+    null_sample_size : int
+        The number of samples to generate for the bootstrap distribution to approximate the pvalue, by default 1000.
+    threshold : float
+        The threshold set on the value of eigenvalues, by default 1e-5. Used to regularize the method.
+    normalize_data : bool
+        Whether the data should be standardized to unit variance, by default True.
+
+    Notes
+    -----
+    Any callable can be given to create the kernel matrix. For instance, to use a particular kernel from sklearn:
+        kernel_X = func:`sklearn.metrics.pairwise.pairwise_kernels.polynomial`
```

Comment:
WDYT about a short description on what the differences are using the conditional vs marginal independence test?

Might be useful for users who are wondering what's going on internally and I suspect will help document why there is going to be significant runtime differences (since the conditional case has more computation).

Reply:
What kind of difference would you point out? Or you mean something like "Note that the conditional test is significantly slower than the unconditional version"?